### PR TITLE
Initial step towards Spanish support: Tokenizer and simple token to words

### PR DESCRIPTION
**Pull request by zeehio**
This pull request related to issue #3  provides a tokenizer and dummy token to words rules for Spanish.

These dummy token to word rules need to be improved to handle all numbers, recognize common Spanish abbreviations (tel. -> teléfono...), recognize roman numbers in expressions like "siglo XX", etc.

I have been working (not as much as I would like to, my time is very limited lately) on fixing https://github.com/TALP-UPC/saga, a library for phoneme transcription in Spanish that covers seven Spanish variants (Argentina, Castilla -Spain-, Chile, Colombia, Mexico, Perú, Venezuela).

I have found some CC-BY-SA Spanish HTS voice models, built by [aholab](http://aholab.ehu.es/aholab/), the speech processing lab of the University of the Basque Country.

If anyone wants to work on improving the token to words rules feel free to do so. Bringing HTS support into mimic would probably be a desirable feature too (#76) but it may be a bit harder to do.

### CODE REVIEW

File: unittests/es_tokenstream_test_main.c [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+}
+
+TEST_LIST =
+{
+    {
+    "Spanish token to words", test_es_tokenizer}
+    ,
+    {
+    0}
+};
```
Comment (forslund) [inline unittests/es_tokenstream_test_main.c]: Is this list layout intentional or the result of some auto-indentation tool? IMHO it's kind of hard to read.

File: unittests/es_tokenstream_test_main.c [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+        for (it = relation_head(rel); it; it = item_next(it)) {
+            /*printf("name: %s\n", item_feat_string(it, "name"));*/
+            TEST_CHECK(cst_streq(item_feat_string(it, "name"), word_list[j]));
+            j++;
+        }
+        delete_utterance(u);
+    }
+        TEST_CHECK(i == expected_num_tok);      /* verify number of tokens */
+        TEST_CHECK(j == expected_num_words);      /* verify number of words */
+    ts_close(ts);
```
Comment (forslund) [inline unittests/es_tokenstream_test_main.c]: Indentation

**Comment by forslund** [comment]
This looks generally good, I still have to look through the main part of the code for minor nit-picks.

One general thing, should we have this enabled by default? As I see it, it might cause some confusion since it's still not a 100% working implementation of the Spanish language.

**Comment by forslund** [comment]
Ok I think I'm done reviewing this. The only thing blocking the merge is the possible memory leak when using `strdup`.

**Comment by sabas1080** [comment]
Can this already be test?

**Comment by zeehio** [comment]
You cannot hear anything yet. We lack several other pieces.

But there are tests to ensure the implemented features in the tokenizer are working, meaning that the token "17" is transformed to the word "diecisiete" (as text).

**Comment by adocampo** [comment]
>But there are tests to ensure the implemented features in the tokenizer are working, meaning that the token "17" is transformed to the word "diecisiete" (as text).

I've been taking a look to aholab AhoTTS and is pretty amazing. It can transform those ordinals and cardinals and accent and voice is quite correct, to be honest.
The demo is here: http://aholab.ehu.es/tts/ and the source code here: https://sourceforge.net/projects/ahotts/

The problem is my programming skills are very rusty at the moment, and I can't contribute in this way.

**Pull request merged**